src/ir/function.rs

```diff
@@ -451,7 +462,12 @@ impl FunctionSig {
         } else {
             ty.ret_type().ok_or(ParseError::Continue)?
         };
-        let ret = Item::from_ty_or_ref(ty_ret_type, cursor, None, ctx);
+
+        let ret = Item::from_ty_or_ref(
+            ty_ret_type,
+            ty_ret_type.declaration(),
```

Comment:
Hmm, so I see how this works, but it is a bit fishy to pass the declaration here just to match it from NoDeclFound afterwards. I suspect this can break other stuff but I'd need to think about what...

Reply:
You're right about that, I think that this is the source of the test failures below.